Tests fail on Windows

We have a couple issues making tests fail on Windows:

1. Windows can't handle environment variables in line; we need to use something like `cross-env` instead.
2. The tests themselves seem to fail for one of two errors that don't occur on Linux/WSL:
    1. Can't open settings.json:
       ```
         console.warn
         Error: ENOENT: no such file or directory, open 'settings.json'
               at Object.openSync (node:fs:589:3)
               at Object.readFileSync (node:fs:457:35)
               at Object.readFileSync (C:\Temp\fluid-queue\src\settings.ts:27:23)
               at Runtime._execModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1429:24)
               at Runtime._loadModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1013:12)
               at Runtime.requireModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:873:12)
               at Runtime.requireModuleOrMock (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1039:21)
               at Object.<anonymous> (C:\Temp\fluid-queue\src\twitch-api.ts:14:19)
               at Runtime._execModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1429:24)
               at Runtime._loadModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1013:12)
               at Runtime.requireModule (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:873:12)
               at Runtime.requireModuleOrMock (C:\Temp\fluid-queue\node_modules\jest-runtime\build\index.js:1039:21)
               at C:\Temp\fluid-queue\tests\simulation.ts:239:16
               at processTicksAndRejections (node:internal/process/task_queues:95:5)
               at mockTwitchApi (C:\Temp\fluid-queue\tests\simulation.ts:239:10)
               at mockModules (C:\Temp\fluid-queue\tests\simulation.ts:54:22)
               at simRequireIndex (C:\Temp\fluid-queue\tests\simulation.ts:266:5)
               at Object.<anonymous> (C:\Temp\fluid-queue\tests\data-conversion\data-conversion.test.ts:167:17) {
            errno: -4058,
            syscall: 'open',
            code: 'ENOENT',
            path: 'settings.json'
         }

            362 |     handle_func = asMock(chatbot_helper.setup).mock.calls[0][0];
            363 |   } catch (err) {
         > 364 |     console.warn(err);
               |             ^
            365 |     if (err != null && typeof err === "object") {
            366 |       (err as Record<string, unknown>).simIndex = {
            367 |         fs,

            at console.<anonymous> (node_modules/jest-mock/build/index.js:709:23)
            at warn (tests/simulation.ts:364:13)
            at Object.<anonymous> (tests/data-conversion/data-conversion.test.ts:167:17)
       ```
    2. TS1343, which is just bizarre, and I don't know what I did to cause it nor how to fix it:
       ```
       tests/chat-logs/chat-logs.test.ts:374:45 - error TS1343: The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'es2022', 'esnext', 'system', 'node16', or 'nodenext'.`
       ```

In both cases, those errors are repeated many times.

This isn't a *huge* issue, since our CI and our development mostly happens in places where the tests run fine, but we should really be able to test on Windows properly.

--- a/package.json
+++ b/package.json
@@ -31,7 +31,7 @@
     "zod": "^3.21.4"
   },
   "scripts": {
-    "test": "NODE_OPTIONS=--experimental-vm-modules jest --config ./tests/jest.config.json --verbose",
+    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js --config ./tests/jest.config.json --verbose",
     "start": "node --enable-source-maps ./build/index.js",
     "prepare": "node -e \"try { require('husky').install() } catch (e) {if (e.code !== 'MODULE_NOT_FOUND') throw e}\"",
     "build": "node --loader @swc-node/register/esm ./build.ts",

--- a/tests/data-conversion/data-conversion.test.ts
+++ b/tests/data-conversion/data-conversion.test.ts
@@ -600,9 +600,12 @@ test("test-renamed-or-deleted", async () => {
   const index = await simRequireIndex(volume, undefined, 1683058095000);
   const mockFs = index.fs;
   // should load without errors and no exception was thrown
-  const fileName = `data/lost-levels-${
-    new Date().toISOString().replaceAll(":", "").split(".")[0]
-  }Z.json`;
+  const fileName = path.join(
+    "data",
+    `lost-levels-${
+      new Date().toISOString().replaceAll(":", "").split(".")[0]
+    }Z.json`
+  );
   expect(consoleWarnMock).toHaveBeenCalledTimes(4);
   expect(consoleWarnMock).toHaveBeenCalledWith(
     "4 users in your queue could not be found!"

